src/EA.Iws.DataAccess/Repositories/Reports/FreedomOfInformationRepository.cs

```diff
@@ -39,8 +40,26 @@ public FreedomOfInformationRepository(IwsContext context)
                     [ImporterAddress],
                     [FacilityName],
                     [FacilityAddress],
-                    COALESCE([QuantityReceived], 0) AS [QuantityReceived],
-                    [QuantityReceivedUnit],
+                    COALESCE(                    
+                        (SELECT	SUM(
+                            CASE WHEN [QuantityReceivedUnitId] IN (1, 2) -- Tonnes / Cubic Metres
+                                THEN COALESCE([QuantityReceived], 0)
+                            ELSE 
+                                COALESCE([QuantityReceived] / 1000, 0) -- Convert to Tonnes / Cubic Metres
+                            END
+                            ) 
+                            FROM [Reports].[Movements]
```

Comment:
I'm concerned about the performance impact of nested select statements. Is it possible to do this by joining onto the Movements view instead?